packages/monocdk/package.json

```diff
@@ -256,7 +256,7 @@
     "@types/fs-extra": "^8.1.1",
     "@types/node": "^10.17.42",
     "cdk-build-tools": "0.0.0",
-    "constructs": "^3.0.4",
+    "constructs": "^3.2.0",
```

Comment:
This is a braking change for monocdk.

Reply:
added do-not-merge

Reply:
I  was wrong, this is not a breaking change as the version bump is for the `devDependencies` and not for `peerDependencies`. But we should still keep the version in `devDependencies` to be the minimal version we support

Reply:
Yes, this was an oversight that I only bumped the `devDependency` and not the peer. We need to discuss what's the path to allow ourselves to upgrade the minimal peer requirement. Otherwise we are stuck forever in 3.0.4 which does not make sense.

Reply:
To add some background. Peer dependencies are a part of the public API. A consumer `package.json` could include:

```js
dependencies: {
    "monocdk": "^1.69.0",
     "constructs": "3.0.4"  
}
```

And it is a valid configuration. 

If `monocdk`  version `1.70.0` bumps the minimal supported version of `constructs` to `3.2.0` , when trying to upgrade to version `1.70.0` the consumer will break as it must bump the `constructs` version as well. There could be many reason for a consumer to declare a fixed dependency on `constructs. Could be because they are using an experimental API of constructs that was removed from newer versions, or they use a third party dependency that declares a direct dependency on the older version of `constructs`.

Reply:
2 points:

1. Does `constructs` contain APIs that can break backwards compatibility in minor version upgrades? Because I'm pretty sure it can't, for the same reason [mono-CDK can't](https://github.com/aws/aws-cdk-rfcs/blob/master/text/0249-v2-experiments.md#appendix-a---why-cant-we-break-backwards-compatibility-in-the-code-of-mono-cdk-main-modules).

2. I don't think third-party packages should have a direct dependency on `constructs`, they should use peer dependencies as well.

In conclusion: I think it's OK to bump the minimum supported minor version of the `constructs` peer dependency. Not being able to do that would be extremely limiting IMO.

Reply:
It's not about API breaking. It's about not being able to upgrade to a newer version of `monocdk` without upgrading `constructs`.

Reply:
1. That's fine. If there are no backwards-incompatible changes possible in the minor version bump of `constructs`, that means they can always safely update the version of `constructs` that they use.
2. Sure. A third-party library can also depend on `constructs` version `3.0.3`, in which case we have this problem already. For that reason, I don't think it's worth it to spend a lot of time worrying about this particular problem.

Reply:
See https://github.com/aws/aws-cdk/pull/11145#issuecomment-717786931